src/InputManager.ts

```diff
@@ -21,39 +21,24 @@ interface InputSession {
 }
 
 export class InputManager {
-    inputWrapper: HTMLElement;
+    renderOptions: RenderOptions;
     inputMode: InputMode;
     waiting: boolean;
     batchInput: string;
     onSend: () => void;
     session: InputSession;
 
-    inputTextArray?: Uint8Array;
-    inputMetaData?: Int32Array;
-    textEncoder: TextEncoder;
+    channel: Channel;
+    messageId = "";
 
     constructor(onSend: () => void, inputMode: InputMode) {
-        this.inputWrapper = document.getElementById(INPUT_AREA_WRAPPER_ID) as HTMLElement;
         this.inputMode = inputMode;
         this.session = { lineNr: 0 };
         this.batchInput = "";
-        this.textEncoder = new TextEncoder();
-        if (typeof SharedArrayBuffer !== "undefined") {
-            papyrosLog(LogType.Important, "Using SharedArrayBuffers");
-            // shared memory
-            this.inputTextArray = new Uint8Array(
-                new SharedArrayBuffer(Uint8Array.BYTES_PER_ELEMENT * 1024));
-            // 2 Int32s:
-            // index 0 indicates whether data is written
-            // index 1 denotes length of the string
-            this.inputMetaData = new Int32Array(
-                new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT * 2));
-        } else {
-            papyrosLog(LogType.Important, "Using serviceWorker for input");
-        }
+        this.channel = makeChannel({ atomics: {} })!; // by default use Atomics
```

Comment:
```suggestion
        this.channel = makeChannel()!; // by default use Atomics
```

Atomics is always used if it's available. The `atomics/serviceWorker` key is just to separate options into groups, not to specify what to use.

Reply:
I wanted to make it more clear what was being used, but the comment also achieves that I suppose.